xwiki-commons-core/xwiki-commons-observation/xwiki-commons-observation-api/src/main/java/org/xwiki/observation/event/ApplicationEvent.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.observation.event;
+
+/**
+ * All ApplicationEvent types must implement this interface.
+ *
+ * @version $id$ (return the application event id)
+ */
+public interface ApplicationEvent
```

Comment:
As I told you in the issue, you are not supposed to completely replace Event. The point is to have a common interface for ApplicationStartedEvent and ApplicationStoppedEvent but as you saw yourself in the Event javadoc all events have to implement it so ApplicationEvent should extend Event (otherwise it not an event).

Reply:
Oh alright, yes I forgot to extend it. Doing it.

Reply:
BTW I think this shows you haven't executed the build or tested the change since that would clearly break XWiki....

Reply:
@vmassol  I tested using `mvn clean test` , can you suggest better way? I mean I can build but this is not XWiki-Platform where I can build and test the real application.

Reply:
ok it means there's no tests in this module if the build passed. What you can do to check that a change works is to build the module and copy the JAR in an xwiki instance in WEB-INF/lib and restart.

Reply:
@vmassol that's helpful. Thanks.